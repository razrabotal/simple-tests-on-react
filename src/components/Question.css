:root {
  --green: #0cc012;
  --greenLight: #cdeccf;
}

.q-question {
  margin-bottom: 24px;
  position: relative;
}

/* .q-question:before {
  content: "";
  position: absolute;
  top: 10px;
  width: 100%;
  display: block;
  margin-bottom: 7px;
  margin-top: 1px;
  font-size: 24px;
  background: #eee;
  height: 10px;
  border-radius: 5px;
  animation: fade-out 0.25s forwards;
  left: 0;
  right: 0;
  margin: auto;
} */
.q-question {
  position: relative;
  /* opacity: 0; */
  /* margin-top: -10px; */
  /* margin-bottom: 0; */
  /* transform: scale(1,0.98); */
  background: #fff;
  animation: show-que 0.6s ease-out forwards; 
  transform-origin: top center;
  will-change: transform;
  border: 1px solid #cdd0d8;
  box-shadow: 1px 1px 3px 0px rgba(60, 80, 100, 0.05),
    3px 3px 3px -1px rgba(60, 80, 100, 0.05),
    8px 8px 12px -4px rgba(60, 80, 100, 0.05),
    12px 12px 15px -4px rgba(60, 80, 100, 0.05),
    16px 16px 25px -4px rgba(60, 80, 100, 0.05);
  padding: 36px 48px 48px;
  border-radius: 5px;
}
/* .q-question > .q-list{ 
  animation: show-que 0.4s ease-out forwards; 
  transform-origin: bottom left;
  will-change: transform;zz
} */


@keyframes fade-out {
  0% {
    opacity: 1;
   width: 100%;
  }
  50% {
    opacity: 1;
    width: 50%;
  }
  100% {
    opacity: 1;
    width: 0%;
  }
}
 /* .q-question:before {
  content: "";
  position: static;
  top: 0;
  width: 100%;
  display: block;
  margin-bottom: 7px;
  margin-top: 1px;
  font-size: 24px;
  background: #eee;
  height: 10px;
  border-radius: 50px;
}  */
/* 
.q-question-list > *:last-child:after {
  content: "";
  display: block;
  margin-bottom: 7px;
  margin-top: 1px;
  font-size: 24px;
  background: #eee;
  height: 10px;
  border-radius: 50px;
} */

@keyframes show-que {
  0% {
    /* margin-top: -100px; */
    /* top: -100px; */
    transform: scale(0.99,0.99);
    margin-top: -280px;
    /* opacity: 1; */
  }
  50% {
    /* margin-top: -30px; */
    /* top:  -30px; */
    transform: scale(0.99,0.99);
    margin-top: -30px;
    /* opacity: 1; */
    /* transform: scale(1,0.92); */
  }
  100% {
    transform: scale(1,1);
    margin-top: 0;
    /* top: 0; */
    /* opacity: 1; */
    /* transform: scale(1,1); */
  }
}

.q-question-list .q-question:not(:last-child){
  z-index: 10;
}
.q-question-list .q-question:last-child{
  z-index: 1;
}
/* .q-question-list .q-question:first-child {
  transform: scale(1,1)!important;
  opacity: 1!important;
  animation: none;
} */


.q-img {
  /* margin-top: 8px; */
  margin-bottom: 4px;
}
.q-img img {
  width: 100%;
  max-height: 600px;
  object-fit: contain;
  object-position: left center;
}


.q-top {
  display: flex;
  align-items: baseline;
  margin-bottom: 8px;
}
  
.q-title {
  font-size: 24px;
  flex: 1;
  line-height: 1.3;
  /* font-weight: 600; */
}
.q-title-number {
  font-size: 24px;
  display: inline-block;
}
.q-title-number:after {
  content: ".";
  margin-right: 5px;
  margin-left: -1px;
}
/* .q-title-number {
  display: inline-block;
  margin-right: 5px;
  margin-top: -3px;
  font-weight: 400;
  box-shadow: inset 0 0 0 1px rgba(155, 155, 161, 0.5);
  border-radius: 50%;
  width: 26px;
  height: 26px;
  line-height: 25px;
  text-align: center;
  font-size: 14px;
  vertical-align: middle;
  letter-spacing: -0.1em;
} */
.q-title-number--list {
  margin-bottom: 7px;
  margin-top: 1px;
  left: 0;
  font-size: 24px;
  background: #fff;
  height: 50px;
  border-radius: 5px;
  width: 97%;
  margin: 0 auto;
  display: block;
  border: 1px solid #ddd;
  box-shadow: 1px 1px 3px 0px rgba(60, 80, 100, 0.04),
  3px 3px 3px -1px rgba(60, 80, 100, 0.02),
  8px 8px 12px -4px rgba(60, 80, 100, 0.02),
  12px 12px 15px -4px rgba(60, 80, 100, 0.02),
  16px 16px 25px -4px rgba(60, 80, 100, 0.01);
  margin-top: -44px;
}

.q-title-number--list:nth-last-child(1) {
  width: 99%;
}
.q-title-number--list:nth-last-child(2) {
  width: 98%;
}
/* .q-title-number--list:not(:last-child) {
  margin-right: 8px;
} */
.q-title-number--list:after {
  content: none;
}
  
.q-number {
  font-size: 12px;
  margin-bottom: 16px;
  letter-spacing: 2px;
  line-height: 24px;
}


.q-list {
  display: flex;
  flex-direction: column;
  margin-bottom: 16px;
  padding: 0 0 0 0px!important;
  /* transform: scale(1,0.85); */
  /* will-change: transform;
  transform-origin: center top; */
  /* animation: show-list 0.15s forwards; */
}
@keyframes show-list {
  to {
    transform: scale(1,1);
  }
}
.q-question-list > .q-question:first-child {
  animation: none;
  opacity: 1;
  transform: scale(1,1)!important;
}
.q-list-item {
  position: relative;
  
  margin: 0!important;
}


.q-button {
  display: flex;
  align-items: center;
  position: relative;
  border: 0;
  line-height: 1.3;
  border-bottom: 2px solid #ddd;
  /* background: #fff; */
  padding: 9px 0 4px 0;
  color: #000;
  font-size: 17px;
  width: 100%;
  text-align: left;
  transition: 0.2s;


  /* justify-content: space-between; */
  align-items: flex-start;
}

.q-button-text {
  position: relative;
  z-index: 2;
  display: block;
}

/* .q-statistics {
  font-size: 80%;
  color: #bbb;
  font-weight: 400;
}
.q-true .q-statistics {
  color: var(--green);
} */

.q-true .q-statistics {
  background: var(--green);
} 
.q-statistics {
  position: absolute;
  bottom: -2px;
  height: 2px;
  left: 0;
  background: #aaa;
  width: 0%;
  transition: 0.5s ease-out;
}

.q-button:hover {
  /* color: var(--green); */
  border-bottom-color: #666;
  cursor: pointer;
}
/* .q-button:hover:before {
  content: "";
  position: absolute;
  width: 5px;
  border-radius: 20px;
  height: 5px;
  background: #000;
  left: -18px;
  top: 18px;
}
.q-select:before, .q-select:hover:before {
  content: none;
} */

.q-false {
  background: #666;
}
  
.q-select, .q-select:hover {
  color: #000;
  border-bottom-color: #eee;
}
.q-true div, .q-true div:hover{
  color: var(--green)!important;
  border-bottom-color: var(--greenLight)!important;
  /* font-weight: 600; */
}
.q-user-select {
  color: #000;
}




.check-mark {
  display: none;
  position: absolute;
  left: -19px;
  top: 12px;
  width: 15px;
  fill:none;
  stroke: #000;
  stroke-linejoin:round;
  stroke-miterlimit:10;
  stroke-dasharray: 20;
  stroke-dashoffset: 20;
}

@keyframes dash {
  to {
    stroke-dashoffset: 0;
  }
}

.q-user-select .check-mark {
  display: block;
  animation: dash 0.25s ease-in forwards;
}
.q-user-select .q-button-text {
  font-weight: 600;
}

.q-true.q-user-select .check-mark {
  stroke:var(--green);
}

.q-comment {
  /* margin-bottom: 16px; */
  /* display: inline-block; */
  position: relative;
  display: block;

  /* background: #faf3de; */
  font-size: 14px;
  color: #000!important;
  font-weight: 400!important;
  /* padding: 4px 10px 5px; */
  /* margin: 8px 0 0; */
  /* transform: scale(0.92,1); */
  animation: show-comment 0.2s forwards;
  transform-origin: top left;
  will-change: transform;
  /* border: 1px solid #ccc; */
  line-height: 1.3;
  /* margin-top: 6px; */
  /* z-index: -1; */
  /* margin-top: -8px;
  margin-bottom: -6px;
  border-radius: 0 5px 5px 5px; */
}
.q-comment::before {
  /* content: "â€” "; */
}
 .q-comment::before {
  /* content: '';
  position: absolute; */
  /* left: -1px;
  top: -8px; */
  /* left: -8px;
  top: -1px;
  width: 9px;
  height: 20px;
  background: url("data:image/svg+xml,%3Csvg width%3D%229px%22 height%3D%2220px%22 viewBox%3D%220 0 9 20%22 xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A  %3Cg fill%3D%22none%22%3E%0A    %3Cpath fill%3D%22%23fff%22 d%3D%22M1.29%2C0 L9%2C0 L9%2C20 L7%2C20 L7%2C17.411 C7%2C14.298 6.413 11.233 5.24 8.218 C4.336 5.893 2.794 3.733 0.614 1.738 L0.614 1.738 C0.207 1.365 0.179 0.732 0.552 0.325 C0.741 0.118 1.009 0 1.29 0 Z%22%3E%3C%2Fpath%3E%0A    %3Cpath stroke%3D%22%23ccc%22 stroke-width%3D%221%22 d%3D%22M9%2C0.5 L1.29%2C0.5 C1.149%2C0.5 1.015 0.559 0.921 0.662 C0.734%2C0.866 0.748 1.182 0.952 1.369 C3.186%2C3.414 4.772 5.637 5.706 8.036 C6.902%2C11.109 7.5 14.235 7.5 17.411 L7.5%2C20%22%3E%3C%2Fpath%3E%0A  %3C%2Fg%3E%0A%3C%2Fsvg%3E%0A") center no-repeat;
  pointer-events: none; */
  /* transform: scale(1,-1) rotate(-90deg);
  transform-origin: top left; */
} 


@keyframes show-comment {
  0% {
    transform: scale(1,0.96);
  }
  100% {
    transform: scale(1,1);
  }
}
.q-true .q-comment, .q-true .q-comment:hover {
  color: #000!important;
  border-bottom-color: #ccc!important;
}