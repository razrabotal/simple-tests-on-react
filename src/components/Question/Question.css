:root {
  --green: #0cc012;
  --greenLight: #cdeccf;
}

.q-question {
  position: relative;
}

/* If Display mode is LIST_REVERSE */
.q-question-list-anim {
  opacity: 0;
  max-height: 0;
  padding-bottom: 0;
  will-change: opacity, max-height;
  transform-origin: top left;
}
.q-question-list-anim:first-child {
  animation: show-tr 0.75s 0.5s ease-out forwards; 
}
.q-question:not(:first-child) {
  opacity: 1!important;
  max-height: 100vh!important;
  padding-bottom: 40px!important;
}
@keyframes show-tr {
  0% {
    opacity: 0;
    max-height: 0;
    padding-bottom: 0;
  }
  100% {
    opacity: 1;
    max-height: 700px;
    padding-bottom: 40px;
  }
}

/* If Display mode is SINGLE */
.q-question-single-anim {
  opacity: 0;
  will-change: opacity, transform;
}
.q-question-single-anim:first-child {
  max-height: initial!important;
  animation: show-single 0.25s ease-out forwards; 
}
@keyframes show-single {
  0% {
    opacity: 0;
    transform: translateX(30px);
  }
  100% {
    opacity: 1;
    transform: translateX(0);
  }
}


.q-img {
  margin-bottom: 4px;
}
.q-img img {
  width: 100%;
  max-height: 400px;
  background: #eee;
  object-fit: contain;
  object-position: center center;
}


.q-top {
  max-width: 600px;
  display: flex;
  align-items: flex-start;
  margin-bottom: 4px;
}
  
.q-title {
  font-size: 18px;
  flex: 1;
  line-height: 1.2;
  font-weight: 600;
}
.q-title-number {
  display: inline-block;
}
.q-title-number:after {
  content: ".";
  margin-right: 5px;
  margin-left: -1px;
}

.q-title-number--list {
  margin-bottom: 7px;
  margin-top: 1px;
  font-size: 24px;
  background: #ccc;
  height: 10px;
  border-radius: 50px;
  width: 10px;
}
.q-title-number--list:not(:last-child) {
  margin-right: 8px;
}
.q-title-number--list:after {
  content: none;
}
  
.q-number {
  font-size: 12px;
  margin-bottom: 16px;
  letter-spacing: 2px;
  line-height: 24px;
}


.q-list {
  max-width: 600px;
  display: flex;
  flex-direction: column;
  padding: 0 0 16px 0px!important;
  transform: scale(1,0.85);
  will-change: transform;
  transform-origin: center top;
  animation: show-list 0.15s forwards;
}
@keyframes show-list {
  to {
    transform: scale(1,1);
  }
}
.q-question-list > .q-question:first-child .q-list{
  transform: scale(1,1)!important;
}
.q-list-item {
  position: relative;
  
  margin: 0!important;
}


.q-button {
  font-family: Georgia;
  display: flex;
  align-items: center;
  position: relative;
  border: 0;
  line-height: 1.25;
  border-bottom: 2px solid #ddd;
  padding: 10px 0 4px 0;
  color: #000;
  font-size: 19px;
  width: 100%;
  text-align: left;
  transition: 0.2s;
  align-items: flex;
}

.q-button-text {
  position: relative;
  z-index: 2;
  display: block;
}


.q-true .q-statistics {
  background: var(--green);
} 
.q-statistics {
  position: absolute;
  bottom: -2px;
  height: 2px;
  left: 0;
  background: #aaa;
  width: 0%;
  transition: 0.35s ease-out;
}

.q-button:hover {
  border-bottom-color: #000;
  cursor: pointer;
}

.q-false {
  background: #666;
}
  
.q-select, .q-select:hover {
  color: #000;
  border-bottom-color: #eee;
}
.q-true div, .q-true div:hover{
  transition: 0.15s;
  color: var(--green)!important;
  border-bottom-color: var(--greenLight)!important;
}
.q-user-select {
  color: #000;
}


.check-mark {
  display: none;
  position: absolute;
  left: -20px;
  top: 12px;
  width: 16px;
  fill:none;
  stroke: #000;
  stroke-linejoin:round;
  stroke-miterlimit:10;
  stroke-dasharray: 20;
  stroke-dashoffset: 20;
}

@keyframes dash {
  to {
    stroke-dashoffset: 0;
  }
}

.q-user-select .check-mark {
  display: block;
  animation: dash 0.2s ease-in forwards;
}

.q-true.q-user-select .check-mark {
  stroke:var(--green);
}

.q-comment {
  display: inline-block;
  position: relative;
  font-size: 16px;
  color: #000!important;
  font-weight: 400!important;
  padding: 10px 0 10px;
  transform: scale(0.92,1);
  transform-origin: top left;
  will-change: transform;
  line-height: 1.4;
}

 .q-comment::before {
  content: none;
  position: absolute;
  left: -1px;
  top: -8px;
  width: 9px;
  height: 20px;
  background: url("data:image/svg+xml,%3Csvg width%3D%229px%22 height%3D%2220px%22 viewBox%3D%220 0 9 20%22 xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A  %3Cg fill%3D%22none%22%3E%0A    %3Cpath fill%3D%22%23fff%22 d%3D%22M1.29%2C0 L9%2C0 L9%2C20 L7%2C20 L7%2C17.411 C7%2C14.298 6.413 11.233 5.24 8.218 C4.336 5.893 2.794 3.733 0.614 1.738 L0.614 1.738 C0.207 1.365 0.179 0.732 0.552 0.325 C0.741 0.118 1.009 0 1.29 0 Z%22%3E%3C%2Fpath%3E%0A    %3Cpath stroke%3D%22%23ccc%22 stroke-width%3D%221%22 d%3D%22M9%2C0.5 L1.29%2C0.5 C1.149%2C0.5 1.015 0.559 0.921 0.662 C0.734%2C0.866 0.748 1.182 0.952 1.369 C3.186%2C3.414 4.772 5.637 5.706 8.036 C6.902%2C11.109 7.5 14.235 7.5 17.411 L7.5%2C20%22%3E%3C%2Fpath%3E%0A  %3C%2Fg%3E%0A%3C%2Fsvg%3E%0A") center no-repeat;
  pointer-events: none;
  transform: scale(1,-1) rotate(-90deg);
  transform-origin: top left;
} 


@keyframes show-comment {
  0% {
    transform: scale(0.92,1);
  }
  100% {
    transform: scale(1,1);
  }
}
.q-true .q-comment, .q-true .q-comment:hover {
  color: #000!important;
  border-bottom-color: #ccc!important;
}