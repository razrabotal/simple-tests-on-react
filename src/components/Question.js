import React from 'react';

import './Question.css';

class Question extends React.Component{
	constructor(props){
		super(props);
      
		this.state = { 
      que: props.next2,
      site: 'lol',
      comment: '',
      
      selectItem: '',
      selected: this.props.selected,
      showPercent: false
    };
    
	}
  
  onChanged(e, comment, correct, id) {    
    if(this.state.selected === false) {
      this.setState({
        selected: true,
        selectItem: id,
        comment: comment,
        showPercent: true
      });

      // id начинается не с 0
      id = id - 1;

      var statistics = this.props.statistics || [];
      statistics[id] = +statistics[id] + 1;
      
      
      this.props.iteration(e, correct, this.props.index, statistics);
    }

  }

 
  render() {
    var that = this;
    var que = this.state.que;

    var statistics = this.props.statistics || [];
     
    var getStatistics = index => {
      var sum = 0;
      for(let item of statistics) {
        sum += +item;
      }
      if(this.state.showPercent) {
        return {
          "width": Math.round((statistics[index] || 0) / sum * 100) + "%"
        }  
      } else {
        return {};
      }  
    }

    var showComment = (id) => (this.state.selectItem === id && this.state.comment) ? 
      <div className="q-comment">{this.state.comment}</div> : '';

    // var showComment = () => (this.state.comment) ? 
    //   <div className="q-comment">{this.state.comment}</div> : '';


    var qList = que.an.map(function(item,index){
   
      var qTrue = that.state.selected && item.correct ? 'q-true' : '',
          qSelect = (that.state.selected) ? 'q-select' : '',
          qUserSelect = that.state.selectItem === item.id ? 'q-user-select' : '';
      
      const classesButton = `q-button ${qSelect}`,
            classesLi = `q-list-item ${qUserSelect} ${qTrue}`;
            
      return <li key={index} className={classesLi}>
          
          <div       
            className={classesButton}
            onClick={e => that.onChanged(e, item.comment, item.correct, item.id)}
          >
            <svg className="check-mark" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 5.86 6.29"><path d="M.3,3.73,2.67,5.47A17.73,17.73,0,0,1,5.47.31"/></svg>
            <span className="q-button-text">{item.text} {showComment(item.id)}</span>
            
            <span className="q-statistics" style={ getStatistics(index) }></span>
          </div>
          
      </li>; 
     })
    
    var imgRender = (que.img) ? <div className="q-img"><img alt="" src={que.img}/></div> : '';
      
    var showQuestionText = () => this.state.que.q;
    var showNumberQuestion = () => <div className="q-title-number">{this.state.que.id}</div>;

    

    return (
      <div className="q-question">
           
           {imgRender}
        
        <div className="q-top">

          

          <div className="q-title">
          { showNumberQuestion() }
            { showQuestionText() }
          </div>   
          {/* <div className="q-number">{this.state.que.id}/{this.props.length}</div> */}
        </div>

        
        
        <div className="q-list">
          { qList }
        </div>
        
        {/* {showComment()} */}
        {/* <div className="q-comment">{this.state.comment}</div> */}
      </div>
    );
   }
}

export default Question;
