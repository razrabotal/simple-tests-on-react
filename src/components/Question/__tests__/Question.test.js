import React from 'react';
import ReactDOM from 'react-dom';
import Question from '../Question';

import renderer from 'react-test-renderer';

import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const testQuestion = {
  "index": 3,
  "question": {
    "id": "1",
    "img": "              https://cdn-images-1.medium.com/max/2000/1*cvlXS1ag-ndn5YujaJUQ2w.png",
    "q": "Какое главное правило первой помощи?",
    "an": [
      {
        "id": "1",
        "correct": true,
        "comment": "Отличное начало. Это прописано практически во всех современных стандартах.",
        "text": "Безопасность спасателя превыше всего"
      },
      {
        "id": "2",
        "comment": "Нет, так пострадавших будет больше.",
        "text": "Сам погибай, а товарища выручай"
      },
      {
        "id": "3",
        "comment": "Не совсем. Современные стандарты первой помощи таковы, что навредить вы, скорее всего, не сможете.",
        "text": "Не навреди"
      }
    ]
  },
  "iteration": "[function bound currentIteration]",
  "statistics": [
    22,
    12,
    3
  ]
};

test('Question changes the onVote after click', () => {
  const question = shallow(
    <Question key={testQuestion.index} 
      index={1}
      question={testQuestion.question} 
      statistics={testQuestion.statistics} />
  );

  expect(question.state().selected).toEqual(false);
  question.find('.q-button').first().simulate('click');
  expect(question.state().selected).toEqual(true);
});

it('renders correctly', () => {
  const question = renderer.create(
    <Question key={testQuestion.index} index={1} question={testQuestion.question} statistics={testQuestion.statistics} />
  ).toJSON();
  expect(question).toMatchSnapshot();
});
