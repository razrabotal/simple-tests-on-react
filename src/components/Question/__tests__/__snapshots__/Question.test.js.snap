// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="q-question"
>
  <div
    className="que-container"
  >
    <div
      className="que-container-left"
    >
      <div
        className="q-top"
      >
        <div
          className="q-title"
        >
          Какое главное правило первой помощи?
        </div>
      </div>
      <div
        className="q-list"
      >
        <li
          className="
                      q-list-item 
                      false 
                      false"
        >
          <div
            className="
                        q-button 
                        false"
            onClick={[Function]}
          >
            <svg
              className="check-mark"
              viewBox="0 0 5.86 6.29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.3,3.73,2.67,5.47A17.73,17.73,0,0,1,5.47.31"
              />
            </svg>
            <span
              className="q-button-text"
            >
              Безопасность спасателя превыше всего
            </span>
            <span
              className="q-statistics"
              style={Object {}}
            />
          </div>
        </li>
        <li
          className="
                      q-list-item 
                      false 
                      false"
        >
          <div
            className="
                        q-button 
                        false"
            onClick={[Function]}
          >
            <svg
              className="check-mark"
              viewBox="0 0 5.86 6.29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.3,3.73,2.67,5.47A17.73,17.73,0,0,1,5.47.31"
              />
            </svg>
            <span
              className="q-button-text"
            >
              Сам погибай, а товарища выручай
            </span>
            <span
              className="q-statistics"
              style={Object {}}
            />
          </div>
        </li>
        <li
          className="
                      q-list-item 
                      false 
                      false"
        >
          <div
            className="
                        q-button 
                        false"
            onClick={[Function]}
          >
            <svg
              className="check-mark"
              viewBox="0 0 5.86 6.29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.3,3.73,2.67,5.47A17.73,17.73,0,0,1,5.47.31"
              />
            </svg>
            <span
              className="q-button-text"
            >
              Не навреди
            </span>
            <span
              className="q-statistics"
              style={Object {}}
            />
          </div>
        </li>
      </div>
      
    </div>
    <div
      className="que-container-right"
    >
      <div
        className="q-img"
      >
        <img
          alt=""
          src="              https://cdn-images-1.medium.com/max/2000/1*cvlXS1ag-ndn5YujaJUQ2w.png"
        />
      </div>
    </div>
  </div>
</div>
`;
